go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -263,18 +264,27 @@ func (rs *rowStreamer) buildSelect() (string, error) {
 		}
 		prefix = ", "
 	}
-	buf.Myprintf(" from %v", sqlparser.NewIdentifierCS(rs.plan.Table.Name))
+	// If we know the index name that we should be using then tell MySQL
+	// to use it if possible. This helps to ensure that we are able to
+	// leverage the ordering from the index itself and avoid having to
+	// do a FILESORT of all the results. This index should contain all
+	// of the PK columns which are used in the ORDER BY clause below.
+	var indexHint string
+	if st.PKIndexName != "" {
+		indexHint = fmt.Sprintf(" force index (`%s`)", st.PKIndexName)
```

Comment:
We might want to escape the index name properly before putting it into `fmt.Sprintf`. Its name just might have backticks.

Reply:
Will do. Thanks!